tests/test_fluentd.py

```diff
@@ -1,73 +1,55 @@
-import unittest
-import logging
 from nose.plugins.attrib import attr
-logger = logging.getLogger(__file__)
 
-from tests.common import load_check
+from checks import AgentCheck
+from tests.common import AgentCheckTest
+
 
 @attr(requires='fluentd')
-class TestFluentd(unittest.TestCase):
+class TestFluentd(AgentCheckTest):
+    CHECK_NAME = 'fluentd'
+    CHECK_GAUGES = ['retry_count', 'buffer_total_queued_size', 'buffer_queue_length']
 
-    def test_fluentd(self):
-        config = {
-            "init_config": {
-            },
+    def __init__(self, *args, **kwargs):
+        AgentCheckTest.__init__(self, *args, **kwargs)
+        self.config = {
             "instances": [
                 {
                     "monitor_agent_url": "http://localhost:24220/api/plugins.json",
-                    "plugin_ids": [ "plg1" ],
+                    "plugin_ids": ["plg1"],
                 }
             ]
         }
 
-        agentConfig = {
+        self.agentConfig = {
             'version': '0.1',
             'api_key': 'toto'
         }
 
-        check = load_check('fluentd', config, agentConfig)
-        check.run()
-        metrics = check.get_metrics()
-        for m in metrics:
-            if m[0] == 'fluentd.forward.retry_count':
-                self.assertEquals(m[2], 0)
-            elif m[0] == 'fluentd.forward.buffer_queue_length':
-                self.assertEquals(m[2], 0)
-            elif m[0] == 'fluentd.forward.buffer_total_queued_size':
+    def test_fluentd(self):
+        self.run_check(self.config, agent_config=self.agentConfig)
+        self.assertServiceCheckOK(self.check.SERVICE_CHECK_NAME,
+                                  tags=['fluentd_host:localhost', 'fluentd_port:24220'])
+        for m in self.metrics:
```

Comment:
I don't think it's a good way to test the metrics: if there is no metric, the test will pass.